runtime/vara/src/migrations.rs

```diff
@@ -16,7 +16,80 @@
 // You should have received a copy of the GNU General Public License
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
-#[allow(unused)]
+use core::marker::PhantomData;
+
 use crate::*;
+use frame_support::{
+    pallet_prelude::StorageVersion,
+    traits::{GetStorageVersion, OnRuntimeUpgrade},
+};
+use sp_runtime::traits::Get;
+
+pub struct NominationPoolsMigrationV4OldPallet;
+impl Get<Perbill> for NominationPoolsMigrationV4OldPallet {
+    fn get() -> Perbill {
+        Perbill::from_percent(10)
+    }
+}
+
+pub struct UpdatePalletsVersions<T>(PhantomData<T>);
+
+impl<
+        T: pallet_multisig::Config
+            + pallet_nomination_pools::Config
+            + pallet_bounties::Config
+            + pallet_election_provider_multi_phase::Config,
+    > OnRuntimeUpgrade for UpdatePalletsVersions<T>
+{
+    fn on_runtime_upgrade() -> Weight {
+        // pallet_multisig
+        let onchain = pallet_multisig::Pallet::<T>::on_chain_storage_version();
+        if onchain == 0 {
+            log::info!("pallet_multisig onchain: {:?}", onchain);
+            StorageVersion::new(1).put::<pallet_multisig::Pallet<T>>();
+        }
+
+        let onchain = pallet_multisig::Pallet::<T>::on_chain_storage_version();
+        log::info!("pallet_multisig onchain: {:?}", onchain);
+
+        // pallet_nomination_pools
+        let onchain = pallet_nomination_pools::Pallet::<T>::on_chain_storage_version();
+        if onchain == 0 {
+            log::info!("pallet_nomination_pools onchain: {:?}", onchain);
+            StorageVersion::new(5).put::<pallet_nomination_pools::Pallet<T>>();
+        }
+
+        let onchain = pallet_nomination_pools::Pallet::<T>::on_chain_storage_version();
+        log::info!("pallet_nomination_pools onchain: {:?}", onchain);
+
+        // pallet_election_provider_multi_phase
+        let onchain = pallet_election_provider_multi_phase::Pallet::<T>::on_chain_storage_version();
+        if onchain == 0 {
+            log::info!(
+                "pallet_election_provider_multi_phase onchain: {:?}",
+                onchain
+            );
+            StorageVersion::new(1).put::<pallet_election_provider_multi_phase::Pallet<T>>();
+        }
+
+        let onchain = pallet_election_provider_multi_phase::Pallet::<T>::on_chain_storage_version();
+        log::info!(
+            "pallet_election_provider_multi_phase onchain: {:?}",
+            onchain
+        );
+
+        // pallet_bounties
+        StorageVersion::new(4).put::<pallet_bounties::Pallet<T>>();
+
+        Weight::zero()
+    }
+}
 
-pub type Migrations = ();
+/// All migrations that will run on the next runtime upgrade.
+///
+/// Should be cleared after every release.
+pub type Migrations = (
+    // unreleased
+    UpdatePalletsVersions<Runtime>,
+    pallet_offences::migration::v1::MigrateToV1<Runtime>,
```

Comment:
offences on-chain is 0 so we don't need to correct it in UpdatePalletsVersions, correct?

Reply:
Offences should be 0.